internal/project/watch.go

```diff
@@ -60,3 +64,263 @@ func (w *watchedFiles[T]) update(ctx context.Context, newData T) (updated bool,
 	w.watcherID = watcherID
 	return true, nil
 }
+
+func createGlobMapper(host ProjectHost) func(data map[tspath.Path]string) []string {
+	rootDir := host.GetCurrentDirectory()
+	rootPath := tspath.ToPath(rootDir, "", host.FS().UseCaseSensitiveFileNames())
+	rootPathComponents := tspath.GetPathComponents(string(rootPath), "")
+	isRootWatchable := canWatchDirectoryOrFile(rootPathComponents)
+
+	return func(data map[tspath.Path]string) []string {
+		start := time.Now()
+
+		// dir -> recursive
+		globSet := make(map[string]bool)
+
+		for path, fileName := range data {
+			w := getDirectoryToWatchFailedLookupLocation(
+				fileName,
+				path,
+				rootDir,
+				rootPath,
+				rootPathComponents,
+				isRootWatchable,
+				rootDir,
+				true,
+			)
+			if w == nil {
+				continue
+			}
+			globSet[w.dir] = globSet[w.dir] || !w.nonRecursive
+		}
+
+		globs := make([]string, 0, len(globSet))
+		for dir, recursive := range globSet {
```

Comment:
In strada there is code to watch "packageDir" if it exits and dir otherwise so we can handle the symlinks .. a note here to handle package symlinks later would be good

Reply:
Ah, I should have added that, yeah. Will follow up with that.

Reply:
I was dogfooding `tsgo` so couldn't find all refs on it; is it just `packageDir ?? dir`?